activerecord/lib/active_record/associations/builder/collection_association.rb

```diff
@@ -66,17 +66,5 @@ def #{name.to_s.singularize}_ids=(ids)
         end
       CODE
     end
-
-    def self.wrap_scope(scope, mod)
```

Comment:
I see. This PR changes the`reflection.constraints` result:

Before:

```ruby
Post.reflect_on_association(:comments).constraints
# => [#<Proc:0x007f89c1d5f488@~/lib/active_record/associations/builder/collection_association.rb:78>]
```

After:

```ruby
Post.reflect_on_association(:comments).constraints
# => []
```

But the association is only specified extension block, doesn't be specified a scope. Should `reflection.constraints` return the scope in that case?

https://github.com/rails/rails/blob/v5.1.0.rc2/activerecord/test/models/post.rb#L50

And `HasManyReflection` is `:nodoc:` class. Should we test the `reflection.constraints` result?